src/Lykke.Job.CandlesHistoryWriter.Services/Candles/CandlesCacheInitalizationService.cs

```diff
@@ -51,16 +51,16 @@ public class CandlesCacheInitalizationService : ICandlesCacheInitalizationServic
             await _log.WriteInfoAsync(nameof(CandlesCacheInitalizationService), nameof(InitializeCacheAsync), null, "All candles history is cached");
         }
 
-        private async Task CacheAssetPairCandlesAsync(IAssetPair assetPair, DateTime toDate)
+        private async Task CacheAssetPairCandlesAsync(IAssetPair assetPair, DateTime now)
         {
             await _log.WriteInfoAsync(nameof(CandlesCacheInitalizationService), nameof(InitializeCacheAsync), null, $"Caching {assetPair.Id} candles history...");
 
             foreach (var priceType in Constants.StoredPriceTypes)
             {
                 foreach (var timeInterval in Constants.StoredIntervals)
                 {
-                    var alignedToDate = toDate.TruncateTo(timeInterval);
-                    var alignedFromDate = alignedToDate.AddIntervalTicks(-_amountOfCandlesToStore, timeInterval);
+                    var alignedToDate = now.TruncateTo(timeInterval).AddIntervalTicks(1, timeInterval);
+                    var alignedFromDate = alignedToDate.AddIntervalTicks(-_amountOfCandlesToStore - 1, timeInterval);
```

Comment:
If _amountOfCandlesToStore == 2000, we will have 2001 as an amount of intervals to cache (including the currently 'not-closed' candle), wich exceeds the setting's value. I suppose, there should be no additional 'minus 1' here.

Reply:
Since alignedFromDate is calculated from alignedToDate, which `has +1 tick, -1 is necessary. You can, remove -1 here and run tests, to se how results are affected